From ce9690cd2be087b58229afa18f1c6b5307e21c35 Mon Sep 17 00:00:00 2001
From: Hongcheng Xie <hongcheng.xie@intel.com>
Date: Fri, 11 Sep 2020 14:37:54 +0800
Subject: [PATCH] Force to use OMX codec

Codec2 will not working if host don't support ION device, add a property
to control foce to use OMX codec or not.

Change-Id: I521c66e0a0e3e980c30de4dea74d01cc062d27d5
Tracked-On: ACP-1285
Signed-off-by: Hongcheng Xie <hongcheng.xie@intel.com>
---
 media/codec2/hidl/client/client.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/media/codec2/hidl/client/client.cpp b/media/codec2/hidl/client/client.cpp
index 7e4352d2d5..f07a943b4e 100644
--- a/media/codec2/hidl/client/client.cpp
+++ b/media/codec2/hidl/client/client.cpp
@@ -913,6 +913,14 @@ std::vector<std::string> const& Codec2Client::GetServiceNames() {
                     LOG(INFO) << stringOutput.str();
                 }
 
+                bool forceUseOmx = ::android::base::GetBoolProperty(
+                      "debug.codec.use.omx", false);
+                if (!names.empty() && forceUseOmx) {
+                   // Still use OMX even codec2 service available due to some reason such as no ION device
+                   LOG(WARNING) << "Still use OMX even Codec2 services available.";
+                   names.clear();
+                }
+
                 return names;
             }
             LOG(ERROR) << "Could not retrieve the list of service instances of "
-- 
2.17.1

